.github/workflows/windows.yml

```diff
@@ -0,0 +1,91 @@
+#  SPDX-License-Identifier: GPL-2.0-or-later
+
+name: "Windows builds"
+on: push
+jobs:
+
+  build_msvc_debug:
+    name: "MSVC 14 debug (win-2019)"
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v1
+      - name: "Install packages"
+        shell: pwsh
+        run: |
+          vcpkg install libpng sdl1 sdl1-net opusfile speexdsp
+          vcpkg integrate install
+      - name: "Build"
+        shell: pwsh
+        env:
+          PATH: '${env:PATH};C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\amd64'
+        run: |
+          # build steps
+          cd visualc_net
+          MSBuild -m dosbox.sln -p:Configuration=Debug
+
+  build_msvc_release:
+    name: "MSVC 14 optimized (win-2019)"
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v1
+      - name: "Install packages"
+        shell: pwsh
+        run: |
+          vcpkg install libpng sdl1 sdl1-net opusfile speexdsp
+          vcpkg integrate install
+      - name: "Build"
+        shell: pwsh
+        env:
+          PATH: '${env:PATH};C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\amd64'
+        run: |
+          # build steps
+          cd visualc_net
+          MSBuild -m dosbox.sln -p:Configuration=Release
+
+  build_msys2_gcc_64:
+    name: MSYS2 GCC-9.2 x86_64
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install msys2 environment
+        run: choco install msys2 --no-progress
```

Comment:
I thought msys2 was preinstalled on these machines? I need to double-check that…

Reply:
Only Chocolatey; there unfortunately isn't a canned msys2 VM. 
However, I think that's a good thing for us to take under control - because building a full-featured DOSBox is something that is not explicitly documented such that it's reproducible (users on vogons continue to struggle with windows builds, and are chided or told to follow the docs or "it works for me, you're probably doing something wrong").